CMakeLists.txt

```diff
@@ -80,6 +81,38 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release")
 endif()
 
+# iOS support
+if(CMAKE_SYSTEM_NAME STREQUAL "iOS" )
+  set(ARM ON)
+  # need to ignore this warning for Xcode to be happy
+  list(APPEND ALL_WARNINGS -Wno-shorten-64-to-32;)
+endif()
+
+# ARM support
+if(ARM)
+  # Apple by default has Apple Accelerate. Otherwise fallback to RUY for GEMM
+  if(APPLE)
+    message(STATUS "Using Apple Accelerate SGEMM")
+    option(USE_RUY_SGEMM "Compile with Ruy SGEMM" OFF)
+  else(APPLE)
+    message(STATUS "Using Ruy SGEMM")
+    set(EXT_LIBS ${EXT_LIBS} ruy)
+    option(USE_RUY_SGEMM "Compile with Ruy SGEMM" ON)
+  endif(APPLE)
+
+  # Define that we are using ARM
+  add_compile_definitions(ARM FMA SSE)
+  # Some warnings as errors. I don't feel comfortable about the strict aliasing.
+  set(ARM_WARNINGS "-fno-strict-aliasing -Wno-comment")
+
+  # @TODO this assumes ArmV8. We should also look at armv7
```

Comment:
Can this be problematic if ArmV8 is not available? Will there be a clear warning or error message if that's the case?

Reply:
automatically detecting is a bit tough but I agreee is a good goal. I've made it clear that for now we only support v8 and will work on adding auto archdetect in future so as to not block this PR